src/Components/WebAssembly/WebAssembly.Authentication/src/IRemoteAuthenticationBuilder.cs

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Components.WebAssembly.Authentication;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    /// <summary>
+    /// An interface for configuring remote authentication services.
+    /// </summary>
+    /// <typeparam name="TRemoteAuthenticationState">The remote authentication state type.</typeparam>
+    /// <typeparam name="TAccount">The account type.</typeparam>
+    public interface IRemoteAuthenticationBuilder<TRemoteAuthenticationState, TAccount>
```

Comment:
What's the purpose of these generic type parameters? They don't seem to be used anywhere inside the interface. Could they be removed?

Reply:
We need to flow the generic parameters so that we can guarantee that extension methods that build on top remain type-safe, so for example if you customize the UserFactory to use a custom CustomUserFactory<CustomAccount> you can't call `AddUserFactory<CustomUserFactory<CustomAccount>>` on a `services.AddOidc(options => {...} )` call that is configured to use the base account type.

Reply:
OK, understood. Thanks for clarifying!